code/script.js

```diff
@@ -222,104 +385,189 @@ const generateBoard = () => {
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
 }
 
-// This function to start (and restart) the game
+// This function is to start (and restart) the game
 const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
+  // Setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  // Resets the counter to zero when the game is restarted
+  count = 0
+  // When the game starts the game board is shown, a secret dog is selected and the counter starts
+  generateBoard()
+  setSecret()
+  counter()
 }
 
-// setting the currentQuestion object when you select something in the dropdown
+// Setting the currentQuestion object when the player selects something in the dropdown
 const selectQuestion = () => {
   const category = questions.options[questions.selectedIndex].parentNode.label
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  // These variables stores what option group (category) the question belongs to and stores the actual value of the question we've selected
+  const selectedOption = questions.options[questions.selectedIndex]
+  const value = selectedOption.value
 
   currentQuestion = {
     category: category,
-    // value: value
+    value: value
   }
 }
 
-// This function should be invoked when you click on 'Find Out' button.
+// This function is invoked when you click on 'Find Out' button
 const checkQuestion = () => {
   const { category, value } = currentQuestion
+  let keep
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  // Compares the currentQuestion details with the secret dogs details in a different manner based on category. See if we should keep or remove dogs based on that and then invoke filterCharacters
+  if (category === 'furcolor' || category === 'coat' || category === 'ears') {
+    attribute = category
+    keep = secret[category] === value
+  } else if (category === 'nose' || category === 'clothing' || category === 'accessories') {
+    keep = secret[category].includes(value)
   }
+  filterCharacters(keep)
 }
 
-// It'll filter the characters array and redraw the game board.
+// Filter the characters array and redraw the game board by determining what is the category, and if it's to keep or remove based on the keep variable
 const filterCharacters = (keep) => {
```

Comment:
Good job on changing the game to dogs! Do you think line 435-512 might be good to comment a bit more, just short ones to make eg. next developer looking at this find things a bit easier? Like:
// Fur color
// Ears
etc.?